apps/api/src/app/events/usecases/send-test-email/send-test-email.usecase.ts

```diff
@@ -111,6 +112,10 @@ export class SendTestEmail {
 
       html = data.outputs.body;
       subject = data.outputs.subject;
+
+      if (data.providers && typeof data.providers === 'object') {
```

Comment:
What is the type of `data` from `this.previewStep.execute` right now? I think we can improve the typing such that `data.providers` is always of type `object`, and more specifically it can be `Record<string, unknown>`.

If this needs to happen in the Framework typings, let's do it.